Bottom: e93d2482fbadbab5970e58b2e895145c10480129
Top:    ccdc242c2debe1fcb36d673c1005d1c45760c354
Author: Sean Hefty <sean.hefty@intel.com>
Date:   2013-11-26 13:16:19 -0800

librdmacm: Check 'init' under mutex

ucma_ib_init() does a quick check that access to ibacm has
been initialized.  This check is done outside of the
acm_lock mutex.  We need to check init again inside of
holding the mutex to ensure that we don't run the
initialization code twice.

Signed-off-by: Sean Hefty <sean.hefty@intel.com>


---

diff --git a/src/acm.c b/src/acm.c
index 45bcb81..fa06d31 100644
--- a/src/acm.c
+++ b/src/acm.c
@@ -137,6 +137,9 @@ void ucma_ib_init(void)
 		return;
 
 	pthread_mutex_lock(&acm_lock);
+	if (init)
+		goto unlock;
+
 	if (!ucma_set_server_port())
 		goto out;
 
@@ -155,6 +158,7 @@ void ucma_ib_init(void)
 	}
 out:
 	init = 1;
+unlock:
 	pthread_mutex_unlock(&acm_lock);
 }
